common/components/org.wso2.carbon.data.agents.log4j/src/main/java/org/wso2/carbon/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -122,6 +131,37 @@ protected void append(LoggingEvent event) {
             System.setProperty("javax.net.ssl.trustStorePassword", "wso2carbon");
             isTruststore = true;
         }
+        if (dataPublisher == null) {
+            Pattern pattern = Pattern.compile("(tcp):\\/\\/([a-zA-Z0-9]+):([0-9]+)");
+            Matcher matcher = pattern.matcher(url);
+            if (matcher.find() && authURLSet == null) {
+                authURLSet = "ssl://" + matcher.group(2).toString() + ":" + (
+                        Integer.parseInt(matcher.group(3).toString()) + 100);
+            }
+            try {
+                dataPublisher = new DataPublisher("Thrift", url, authURLSet, userName, password);
```

Comment:
Logger will be in inconsistent state if there is an exception thrown at this line(i.e. dataPublisher== null). We need to handle it properly, may be retry. We should not try to publish events until dataPublisher is ready.